src/_macosx.m

```diff
@@ -629,6 +630,39 @@ static CGFloat _get_device_scale(CGContextRef cr)
     return (PyObject*)self;
 }
 
+int set_icon(PyObject* icon_path) {
+    if (!icon_path) {
+        return 0;
+    }
+    const char* icon_path_ptr = PyUnicode_AsUTF8(icon_path);
+    if (!icon_path_ptr) {
+        Py_DECREF(icon_path);
+        return -1;
+    }
+    NSString* ns_icon_path = [NSString stringWithUTF8String: icon_path_ptr];
+    if (!ns_icon_path) {
+        Py_DECREF(icon_path);
+        PyErr_SetString(PyExc_RuntimeError, "Could not convert to NSString*");
+        return -1;
+    }
+    NSImage* image = [[NSImage alloc] initByReferencingFile: ns_icon_path];
+    [ns_icon_path release];
```

Comment:
I thought you were only supposed to release explicitly alloc'd items?

Reply:
Maybe this is the source of your segfault? I'm not super familiar with Obj-C semantics, but I think you're right at any rate.